File: Inf04styczen2026/mobilna/app/src/main/java/com/github/perzotprogrammer/inf04styczen2026/Kosc.java
package com.github.perzotprogrammer.inf04styczen2026;

import android.graphics.drawable.Drawable;

import java.util.Random;

public class Kosc {
    public static int Instancje = 0;

    public int[] Obrazy = new int[]{
            R.drawable.kosc0,
            R.drawable.kosc1,
            R.drawable.kosc2,
            R.drawable.kosc3,
            R.drawable.kosc4,
            R.drawable.kosc5,
            R.drawable.kosc6,
    };

    public int WyrzuconaWartosc;
    public int ObrazIndeks;
    public boolean CzyDostepna;

    public Kosc(int wyrzuconaWartosc) {
        if (wyrzuconaWartosc < 0 || wyrzuconaWartosc > 6) wyrzuconaWartosc = 0;
        WyrzuconaWartosc = wyrzuconaWartosc;
        ObrazIndeks = wyrzuconaWartosc;
        CzyDostepna = true;
        Instancje++;
    }

    public Kosc() {
        rzut();
        CzyDostepna = true;
        Instancje++;
    }

    public void rzut() {
        Random random = new Random();
        int wylosowana = 1 + (Math.abs(random.nextInt()) % 6);
        WyrzuconaWartosc = wylosowana;
        ObrazIndeks = wylosowana;
    }

    public void blokujKosc() {
        CzyDostepna = false;
    }

    public String wartoscSlownie() {
        switch (WyrzuconaWartosc) {

            case 1:
                return "jeden";
            case 2:
                return "dwa";
            case 3:
                return "trzy";
            case 4:
                return "cztery";
            case 5:
                return "pięć";
            case 6:
                return "sześć";
            default:
                return "nie zdefiniowano";
        }
    }
}

File: Inf04styczen2026/konsolowa/Kosc.cs
namespace wspodzielone;

public class Kosc
{
    public static int Instancje = 0;

    public string[] Obrazy = [
        "kosc0.png",
        "kosc1.png",
        "kosc2.png",
        "kosc3.png",
        "kosc4.png",
        "kosc5.png",
        "kosc6.png"
        ];

    public int WyrzuconaWartosc;
    public int ObrazIndeks;
    public bool CzyDostepna;

    public Kosc(int wyrzuconaWartosc)
    {
        if (wyrzuconaWartosc < 0 || wyrzuconaWartosc > 6) wyrzuconaWartosc = 0;
        WyrzuconaWartosc = wyrzuconaWartosc;
        ObrazIndeks = wyrzuconaWartosc;
        CzyDostepna = true;
        Instancje++;
    }

    public Kosc()
    { 
        CzyDostepna = true;
        Rzut();
        Instancje++;
    }

    public void Rzut()
    {
        if (!CzyDostepna) return;
        Random random = new Random();
        int wylosowana = random.Next(1, 6);
        WyrzuconaWartosc = wylosowana;
        ObrazIndeks = wylosowana;
    }

    public void BlokujKosc()
    {
        CzyDostepna = false;
    }

    public string WartoscSlownie()
    {
        return WyrzuconaWartosc switch
        {

            1 => "jeden",
            2 => "dwa",
            3 => "trzy",
            4 => "cztery",
            5 => "pięć",
            6 => "sześć",
            _ => "nie zdefiniowano",
        };
    }
}

